fearless/Modules/Staking/StakingMain/StakingMainInteractor+InputProtocol.swift

```diff
@@ -3,72 +3,132 @@ import SoraFoundation
 import RobinHood
 
 extension StakingMainInteractor: StakingMainInteractorInputProtocol {
+    func saveNetworkInfoViewExpansion(isExpanded: Bool) {
+        commonSettings.stakingNetworkExpansion = isExpanded
+    }
+
     func setup() {
-        currentAccount = settings.selectedAccount
-        currentConnection = settings.selectedConnection
+        setupSelectedAccountAndChainAsset()
+        setupChainRemoteSubscription()
+        setupAccountRemoteSubscription()
+
+        sharedState.eraValidatorService.setup()
+        sharedState.rewardCalculationService.setup()
```

Comment:
I doubt we will ever setup each service separately, so I'd create `setup()` function inside `sharedState`

Reply:
Or maybe we will... Let's discuss